src/sagemaker/estimator.py

```diff
@@ -282,8 +282,14 @@ def deploy(self, initial_instance_count, instance_type, endpoint_name=None, **kw
     @property
     def model_data(self):
         """str: The model location in S3. Only set if Estimator has been ``fit()``."""
-        return self.sagemaker_session.sagemaker_client.describe_training_job(
-            TrainingJobName=self.latest_training_job.name)['ModelArtifacts']['S3ModelArtifacts']
+        try:
```

Comment:
I think it might be cleaner to check if the entity you want exists rather than catching an `AttributeError`

Reply:
Updated.